File: track-api/Models/Api/ApiDataset.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace track_api.Models.Api
{
    public class ApiDataset
    {
        public int Id { get; set; }

        public string Label { get; set; }

        public TimeSpan Span { get; set; }

        public long Ticks { get; set; }

        public List<int> SeriesIds { get; set; }
        public List<DateTime> SeriesLabels { get; set; }

        public List<ApiSeries> NumericalSeries { get; set; }
        public List<ApiSeries> FrequencySeries { get; set; }
    }
}

File: track-web/src/models/api/ApiDataset.ts
import { ApiSeries } from './ApiSeries';

export class ApiDataset {

  public Id: number;
  
  public Label: string;

  public SeriesIds: number[];
  public SeriesLabels: string[];
  
  public Span: string;
  public Ticks: number;

  public NumericalSeries: ApiSeries[];
  public FrequencySeries: ApiSeries[];

}